### PR TITLE
Add key generation security considerations

### PR DESCRIPTION
**Pull request by wussler**
- Add a key generation paragraph to the security consideration paragraph considering other protocols
- Fix the wording in key generation procedure

### CODE REVIEW

File: draft-ietf-openpgp-pqc.md [hunk 1]
```diff
@@ -1015,6 +1015,11 @@ For the same reasons, this specification further recommends the use of `AES-256`
 This recommendation should be understood as a clear and simple rule for the selection of `AES-256` for encryption.
 Implementations may also make more nuanced decisions.
 
+## Key generation
+
+When generating keys, this specification requires component keys to be generated independently, and recommends not to reuse existing keys for any of the components.
+Note that reusing a key may create a valid message or signature in a different protocol, and reusing a key may open to subtle vulnerabilities.
```
Comment (falko-strenzke) [inline draft-ietf-openpgp-pqc.md:1021]:
```suggestion
Note that reusing a key across different protocols may lead signature confusion vulnerabilities, that formarlly classify as signature forgeries. Generally, reusing a key for different purposes may lead to subtle vulnerabilities.
```

**Review by falko-strenzke — APPROVED** [review]

**Review by TJ-91 — APPROVED** [review]
In `## Key generation strategies` we already mention that keys should be fresh:

> It is RECOMMENDED to generate fresh secrets when generating PQ(/T) keys.

But I think it's ok to repeat that.

**Comment by wussler** [comment]
> In `## Key generation strategies` we already mention that keys should be fresh:
> 
> > It is RECOMMENDED to generate fresh secrets when generating PQ(/T) keys.
> 
> But I think it's ok to repeat that.

This is by design, here it's non-normative and provides a justification

File: draft-ietf-openpgp-pqc.md [hunk 2]
```diff
@@ -1024,7 +1024,7 @@ Implementations may also make more nuanced decisions.
 ## Key generation
 
 When generating keys, this specification requires component keys to be generated independently, and recommends not to reuse existing keys for any of the components.
-Note that reusing a key may create a valid message or signature in a different protocol, and reusing a key may open to subtle vulnerabilities.
+Note that reusing a key across different protocols may lead signature confusion vulnerabilities, that formally classify as signature forgeries. Generally, reusing a key for different purposes may lead to subtle vulnerabilities.
```
Comment (fluppe2) [inline draft-ietf-openpgp-pqc.md:1027]: I think there is a "to" missing in "...may lead signature confusion vulnerabilities..."

Comment (wussler) [reply draft-ietf-openpgp-pqc.md:1027]: Good catch, pushed a fix

**Comment by fluppe2** [comment]
LGTM

**Pull request merged**